ql/src/java/org/apache/hadoop/hive/ql/io/CombineHiveInputFormat.java

```diff
@@ -691,6 +693,18 @@ Map<Path, List<String>> removeScheme(Map<Path, List<String>> pathToAliases) {
   public RecordReader getRecordReader(InputSplit split, JobConf job,
       Reporter reporter) throws IOException {
     if (!(split instanceof CombineHiveInputSplit)) {
+      if (split instanceof FileSplit) {
```

Comment:
won't we break here legacy code? prev if !(split instanceof CombineHiveInputSplit), we fallback to HiveInputFormat#getRecordReader

Reply:
If the conditions are satisfied then we use the `inputFormat` for getting the record reader. Otherwise we do fallback as previously done.

Reply:
👍